exporters/trace/gcp_exporter/internal/recordable_test.cc

```diff
@@ -0,0 +1,115 @@
+#include "exporters/trace/gcp_exporter/recordable.h"
+#include <gtest/gtest.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace gcp
+{
+
+TEST(Recordable, TestSetAttribute)
+{
+    Recordable rec;
+
+    // Set 'bool' type
+    const nostd::string_view bool_key = "bool_key";
+    const common::AttributeValue bool_value = true;
+    rec.SetAttribute(bool_key, std::move(bool_value));
+
+    // Set 'integer' type
+    const nostd::string_view int_key = "int_key";
+    const int64_t seven = 7;
+    const common::AttributeValue int_value = seven;
+    rec.SetAttribute(int_key, std::move(int_value));
+
+    // Set 'string' type
+    const nostd::string_view string_key = "string_key";
+    const common::AttributeValue string_value = "test";
+    rec.SetAttribute(string_key, std::move(string_value));
+
+    auto attr_map = rec.span().attributes().attribute_map();
+    
+    EXPECT_TRUE(attr_map["bool_key"].bool_value());
+    EXPECT_EQ(seven, attr_map["int_key"].int_value());
+    EXPECT_EQ("test", attr_map["string_key"].string_value().value());
+}
+
+TEST(Recordable, TestSetIds)
+{
+    setenv("GOOGLE_CLOUD_PROJECT_ID", "test_project", 1);
+
+    const opentelemetry::trace::TraceId trace_id(
+    std::array<const uint8_t, opentelemetry::trace::TraceId::kSize>(
+    {0, 1, 0, 2, 1, 3, 1, 4, 1, 5, 1, 6, 3, 7, 0, 0}));
+
+    const opentelemetry::trace::SpanId span_id(
+    std::array<const uint8_t, opentelemetry::trace::SpanId::kSize>(
+    {1, 2, 3, 4, 5, 6, 7, 8}));
+
+    const opentelemetry::trace::SpanId parent_span_id(
+    std::array<const uint8_t, opentelemetry::trace::SpanId::kSize>(
+    {4, 5, 0, 1, 1, 1, 1, 3}));
+
+    Recordable rec;
+
+    rec.SetIds(trace_id, span_id, parent_span_id);
+
+    EXPECT_EQ("projects/test_project/traces/0001000201030104/spans/01020304", rec.span().name());
```

Comment:
Yes, I think this is because of the [comment](https://github.com/GoogleCloudPlatform/opentelemetry-operations-cpp/pull/2#discussion_r449032473) above. 
Fixed it.